src/shogun/lib/parameter_observers/ParameterObserverCV.cpp

```diff
@@ -123,14 +110,14 @@ void CParameterObserverCV::print_machine_information(CMachine* machine) const
 	{
 		CLinearMachine* linear_machine = (CLinearMachine*)machine;
 		linear_machine->get_w().display_vector("Learned Weights = ");
-		SG_SPRINT("Learned Bias = %f\n", linear_machine->get_bias())
+		SG_PRINT("Learned Bias = %f\n", linear_machine->get_bias())
```

Comment:
This is not good. The class won’t be available here anymore in the near future. Need to do either visitor pattern or virtual methods but no explicit casting

Reply:
What do you mean here by `won't be available`?

Reply:
Class definition bit within scope. But actually that is not true, sorry.
However I generally preferd if you used the to string method of the respective class, or a new method that prints this info, better style than having all the classes listed in here

Reply:
I guess this will need a bit more of discussion. It may also be a good example of a good entrance task though :)

Reply:
A simple solution would be to add `print_machine_information` to `CMachine`. But then again, we have `to_string` actually. Of course there could be a curated version, for machines that return only the relevant fields. Whatever we do, I think we certainly should not dispatch machine types in here.

Reply:
I think that it would be better to do this kind of task in another PR, since it is somewhat related to refactoring the observers (which it is not related to the `ObservedValue`). Plus, I would like to merge this soon such to keep working on the other tasks.

Reply:
Ok sure, but let’s not forget about it

Reply:
Sure!